SNOW-831868: Fix table name exist check (#878)
snowflakedb/snowpark-python: CHANGELOG.md

- Fixed a bug when `Dataframe.join_table_function` did not run all the necessary queries to setup join table function when sql simplifier was enabled.
- Added comments to SnowflakeFile and added a API reference page for it.

> - Fixed a bug where `DataFrameWriter.save_as_table` and `DataFrame.copy_into_table` fail to parse fully qualified table name.

Reviewer: fail -> failed
table name -> table names

Reviewer 2: @sfc-gh-aling I'll take care of these comments when I merge with release branch. If you push another commit fixing this, we'll rerun tests. I'm trying to avoid that

> - Fixed a bug that `DataFrameWriter.save_as_table` and `DataFrame.copy_into_table` fail to parse table name containing schema name.

Reviewer 3: Maybe:
```suggestion
- Fixed a bug where `DataFrameWriter.save_as_table` and `DataFrame.copy_into_table` fail to parse fully qualified table name.
```
?

Author: updated

  - `bround`.
  - `substring_index`
- Added parameter `skip_upload_on_content_match` when creating UDFs, UDTFs and stored procedures using `register_from_file` to skip uploading files to a stage if the same version of the files are already on the stage.

> - Added support for `DataFrameWriter.save_as_table` method to take table names that contain dots.

Reviewer 4: I'm not sure if we should correct old changelog 🤔

Author: I lean towards to correcting it, so that people won't be confused when they find no `save_as_table` on `DataFrame` when they check our changelog history. this will show up in the default pypi home page and our github main branch
